File: pl/usage.pl
# -*- Perl -*-
# $Id: usage.pl,v 1.15 2000-01-29 04:58:20 satoru Exp $
# Copyright (C) 1997-1999 Satoru Takabayashi All rights reserved.
# Copyright (C) 2000 Namazu Project All rights reserved.
#     This is free software with ABSOLUTELY NO WARRANTY.
#
#  This program is free software; you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation; either versions 2, or (at your option)
#  any later version.
# 
#  This program is distributed in the hope that it will be useful
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with this program; if not, write to the Free Software
#  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
#  02111-1307, USA
#
#  This file must be encoded in EUC-JP encoding
#

package usage;
require "var.pl";

# dummy function.
sub N_ {
    return $_[0];
}

#
# To deceive xgettext, add fake "\n\" at the end of each line.
# This "\n\" is handled in mknmz's show_usage().
#
$USAGE = N_("mknmz %s, an indexer of Namazu.\n\
\n\
Usage: mknmz [options] <target>...\n\
\n\
Target files:\n\
  -a, --all                target all files.\n\
  -e, --robots-txt         exclude files which is excluded by robots.txt.\n\
  -t, --media-type=mtype   set a media type for all target files explicitly.\n\
  -h, --mailnews           same as --media-type='message/rfc822'\n\
      --mhonarc            same as --media-type='text/html; x-type=mhonarc'\n\
  -A, --htaccess           exclude files restricted by .htaccess.\n\
  -F, --target-list=file   load a file which contains list of target files.\n\
      --allow=regex        set a regex for file names which will be allowed.\n\
      --deny=regex         set a regex for file names which will be denied.\n\
      --exclude=regex      set a regex for pathnames which will be excluded.\n\
  -M, --meta               handle HTML meta tags for field-specified search.\n\
  -r, --replace=code       set a code for replacing URI.\n\
      --mtime=int          limit by date, ( minus number for recent, i.e.,\n\
                           -50 for recent 50 days, 50 for older than 50.)\n\
\n\
Morphological Analysis:\n\
  -c, --use-chasen         use ChaSen for analyzing Japanese.\n\
  -k, --use-kakasi         use KAKASI for analyzing Japanese.\n\
  -m, --use-chasen-morph   use ChaSen for collecting only nouns.\n\
\n\
Text Operations:\n\
  -E, --no-edge-symbol     remove symbols on edge of word.\n\
  -G, --no-okurigana       remove Okurigana in word.\n\
  -H, --no-hiragana        ignore words consist of Hiragana only.\n\
  -K, --no-symbol          remove symbols.\n\
\n\
Summarization:\n\
  -U, --no-encode-uri      do not encode URI.\n\
  -x, --no-heading-summary do not make summary with HTML's headings.\n\
\n\
Index Construction:\n\
      --update=index       set an index for updating.\n\
  -Y, --no-delete          do not detect removed documents.\n\
  -Z, --no-update          do not detect update and deleted documents.\n\
\n\
Miscellaneous:\n\
  -s, --checkpoint         turn on the checkpoint mechanism.\n\
  -f, --config=file        set a pathname of a config file.\n\
  -I, --include=file       include your customization file.\n\
  -O, --output-dir=dir     set a directory to output the index.\n\
  -T, --template-dir=dir   set a directory having NMZ.{head,foot,body}.*.\n\
  -q, --quiet              suppress status messages during execution.\n\
  -v, --version            show the version of namazu and exit.\n\
  -V, --verbose            be verbose.\n\
      --debug              be debug mode.\n\
      --help               show this help and exit.\n\
\n\
Report bugs to <%s>.\n\
");

##
## Version information
##
$VERSION_INFO = <<EOFversion;
mknmz of Namazu $var::VERSION
$var::COPYRIGHT

This is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 2, or (at your option)
any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty
of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
EOFversion

1;

File: src/usage.c
/*
 * 
 * usage.c -
 * 
 * $Id: usage.c,v 1.9 1999-11-18 13:42:07 satoru Exp $
 * 
 * Copyright (C) 1997-1999 Satoru Takabayashi  All rights reserved.
 * This is free software with ABSOLUTELY NO WARRANTY.
 * 
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 * 
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 * 
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 * 02111-1307, USA
 * 
 * 
 */
#include <stdio.h>
#include "namazu.h"
#include "codeconv.h"
#include "message.h"
#include "usage.h"

/************************************************************
 *
 * Public functions
 *
 ************************************************************/

/* display the usage and version info and exit */
void show_long_usage(void)
{
    char buf[BUFSIZE * 4];
    char *usage = (char *)
    N_("\
namazu of Namazu %s\n\n\
Usage: namazu [options] <query> [index]... \n\
    -n, --max=num        set number of documents shown at once.\n\
    -w, --whence=num     set first number of documents shown in results.\n\
    -l, --list           print results by listing format.\n\
    -s, --short          print results by short format.\n\
        --results=ext    set NMZ.result.ext for printing results.\n\
        --late           sort documents in late order.\n\
        --early          sort documents in early order.\n\
        --sort=method    set a sort method (score, date, field:name)\n\
        --ascending      sort in ascending order (default: descending)\n\
    -a, --all            print all results.\n\
    -c, --count          print only number of hits.\n\
    -h, --html           print in HTML format.\n\
    -r, --no-references  do not display reference hit counts.\n\
    -H, --page           print further result links. (nearly meaningless)\n\
    -F, --form           force to print <form> ... </form> region.\n\
    -R, --no-replace     do not replace URI string.\n\
    -U, --no-encode-uri  do not decode URI when printing in a plain format.\n\
    -o, --output=file    set output file name.\n\
    -f, --config=file    set a pathname of a config file.\n\
    -C, --show-config    print current configuration.\n\
    -q, --quiet          do not display extra messages except search results.\n\
    -L, --lang=lang      set language (ja or en)\n\
    -v, --version        show the version of namazu and exit.\n\
        --help           show this help and exit\n");

    strcpy(buf, _(usage));
    printf(buf, VERSION);
}


void show_mini_usage(void)
{
    printf(_("Usage: namazu [options] <query> [index]...\n"));
    printf(_("Try `namazu --help' for more information.\n"));
}

void show_version(void)
{
    printf("namazu of Namazu %s\n\n", VERSION);
    printf("Copyright (C) 1997-1999 Satoru Takabayashi All rights reserved.\n");
    printf("This is free software; you can redistribute it and/or modify\n");
    printf("it under the terms of the GNU General Public License as published by\n");
    printf("the Free Software Foundation; either version 2, or (at your option)\n");
    printf("any later version.\n\n");
    printf("This program is distributed in the hope that it will be useful,\n");
    printf("but WITHOUT ANY WARRANTY; without even the implied warranty\n");
    printf("of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the\n");
    printf("GNU General Public License for more details.\n");
}
